### DISCUSSION TITLE
Combine multiple query_filters with logical AND

### DISCUSSION THREAD

**Discussion by philket** [netbox-community/ansible_modules#1263] [Q&A]
Hey,

I read in the documentation that the entries in query_filter are combined with a logical OR. I would like to have them combined with a logical AND instead, so that the Ansible inventory is built only from devices with status active AND a certain tag. Is there any possibility to achieve that?

**Comment by sc68cal** [comment]
I believe you would do this by adding the following to your inventory plugin configuration

```yaml
group_by:
  - tags
query_filters:
  - status: active
```

And then running your ansible-playbook with a `--limit <name_of_your_tag_in_netbox>` or a `hosts: <name of your tag>` in your playbook host specification

You can use `ansible-inventory --list` to see the groups that get generated from your tags in netbox

**Reply by rafaelbianco** [reply]
Hi! I'm just learning Ansible and this is an option that would make tons of sense.

On large enviroments and multiple teams, it would be beneficial to pass queries as a logical and, not or.

Maybe the idea behind is not making ansible go everytime to netbox to get it's inventory (since there cache options for the plugin) but not having the option for an AND filter is a bummer.

In my case, the brilliant part of netbox are the filters. If I can't use AND logic I'm overwhelmed by tons of stuff.

That said, logical AND on the query would also ease the strain of the postgree db.

I would love to see this implemented.

**Reply by rafaelbianco** [reply]
> I believe you would do this by adding the following to your inventory plugin configuration
> 
> ```yaml
> group_by:
>   - tags
> query_filters:
>   - status: active
> ```
> 
> And then running your ansible-playbook with a `--limit <name_of_your_tag_in_netbox>` or a `hosts: <name of your tag>` in your playbook host specification
> 
> You can use `ansible-inventory --list` to see the groups that get generated from your tags in netbox

Quick question.

That query would return any device with status active and would group them by tags.

If you had 10 tags you would end up with 10 gropus, one for each tag, right? (I'm very new to this, sorry if dumb question).

But with your netbox had 10k devices, all active and with tags, your invetory would be 10k devices, right?

Isn't that harmful to the database? Logical AND filters are the default on the GUI if I'm not mistaken, why not also on the on the query filter?
